src/libcore/num/mod.rs

```diff
@@ -4768,14 +4768,29 @@ fn from_str_radix<T: FromStrRadixHelper>(src: &str, radix: u32) -> Result<T, Par
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct ParseIntError {
-    kind: IntErrorKind,
+    pub kind: IntErrorKind,
 }
 
+/// Enum to store the various types of errors that can cause parsing an integer to fail.
+#[unstable(feature = "int_error_matching",
+           reason = "it can be useful to match errors when making error messages \
+                     for integer parsing",
+           issue = "22639")]
 #[derive(Debug, Clone, PartialEq, Eq)]
-enum IntErrorKind {
+#[non_exhaustive]
+pub enum IntErrorKind {
```

Comment:
Maybe the original struct should be replaced with this enum instead of having a `kind` field?

Reply:
Are you saying that `struct ParseIntError` should be removed. Would that not be a breaking change? Can I just try doing that?

Or should I make it into `pub struct ParseIntError(IntErrorKind)`.

Reply:
No, I mean the enum would get the name `ParseIntError`.

Reply:
That is a good idea. I may try that after I get a response to this PR from the libs team.